### PR TITLE
curveadm born

### PR DESCRIPTION
**Pull request by Wine93**

### CODE REVIEW

File: configs/test.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+global:
+  user: "nbs"
+  ssh_port: 1046
+  private_key_file: "~/id_pub"
+  data_dir: "/home/nbs/data"
+  log_dir: "/home/nbs/log"
+  s3_ak:
+  s3_sk:
+  s3_endpoint:
+  s3_bucker_name:
```
[the remark this answers is not in the document]
Comment (Wine93) [reply configs/test.yaml:10]: > s/bucker/bucket/

fixed.

File: cli/command/clean.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+)
+
+type cleanOptions struct {
+	id   string
+	role string
+	host string
+	only []string
+}
+
+func NewCleanCommand(curveadm *cli.CurveAdm) *cobra.Command {
```
Comment (aspirer) [inline cli/command/clean.go:51]: how do this deal with a misoperation? should we add some double check options? just like ceph delete pool cmd: ceph osd pool delete $pool $pool --yes-i-really-mean-it

Comment (aspirer) [reply cli/command/clean.go:51]: or add a interactive, the operator must input a 'YES' to clean a cluster or service?

File: cli/command/cluster/remove.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/opencurve/curveadm/internal/tui/common"
+	cliutil "github.com/opencurve/curveadm/internal/utils/cli"
+	"github.com/spf13/cobra"
+)
+
+type removeOptions struct {
+	clusterName string
+}
+
+func NewRemoveCommand(curveadm *cli.CurveAdm) *cobra.Command {
```
Comment (aspirer) [inline cli/command/cluster/remove.go:39]: the same as the clean cmd. how to deal with misoperation?

File: cli/command/clean.go (hunk 2 above)
Comment (Wine93) [reply cli/command/clean.go:51]: > or add a interactive, the operator must input a 'YES' to clean a cluster or service?

yep, i will improve it.

File: cli/command/cluster/remove.go (hunk 3 above)
Comment (Wine93) [reply cli/command/cluster/remove.go:39]: > the same as the clean cmd. how to deal with misoperation?

i will improve it.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,91 @@
-# curveadm
+# CurveAdm
 Deploy and manage CurveFS cluster
```
Comment (aspirer) [inline README.md:2]: Deploy and manage Curve cluster (only support CurveFS now, CurveBS will be supported later).

File: README.md [hunk 5]
```diff
@@ -1,2 +1,91 @@
...
+
+Requirement
+---
+
+CurveAdm depends on docker, please [install docker](https://docs.docker.com/engine/install/) first.
+
+Please make sure the docker daemon has running, you can run the below command to verify:
+
+```shell
+$ sudo docker run hello-world
```
Comment (aspirer) [inline README.md:23]: add output info for checking.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,91 @@
...
+
+Usage
+---
+
+### Deploy Cluster
+
+Prepare cluster topology, you can refer to the sample configuration:
+
+* [cluster](examples/cluster/services.yaml)
+* [stand-alone](examples/stand-alone/services.yaml)
```
Comment (aspirer) [inline README.md:41]: targets in cluster mode and stand-alone mode are the same 3, stand-alone mode should be 1?

File: README.md (hunk 4 above)
Comment (Wine93) [reply README.md:2]: > Deploy and manage Curve cluster (only support CurveFS now, CurveBS will be supported later).

fixed.

File: README.md (hunk 5 above)
Comment (Wine93) [reply README.md:23]: > add output info for checking.

fixed.

File: examples/stand-alone/services.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  variable:
+    target: 10.0.1.1
+
+etcd_services:
+  config:
+    listen.ip: ${target}
+    listen.port: ${service_sequence}2380  # 12380,22380,32380
+    listen.client_port: ${service_sequence}2379  # 12379,22379,32379
+  deploy:
+    - host: ${target}
```
Comment (aspirer) [inline examples/stand-alone/services.yaml:21]: should be one host here?

Comment (Wine93) [reply examples/stand-alone/services.yaml:21]: > should be one host here?

yep, i will fix it.

File: README.md (hunk 6 above)
Comment (Wine93) [reply README.md:41]: > targets in cluster mode and stand-alone mode are the same 3, stand-alone mode should be 1?

fixed.

**Pull request merged**